src/ngx_http_lua_connection.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (C) 2014-2015 Daurnimator
+ */
+
+/* need to include DDEBUG so that ngx_http_lua_util.h works */
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+#include "ngx_http_lua_connection.h"
+
+#include <math.h> /* HUGE_VAL */
+#include <poll.h> /* POLLIN, POLLOUT */
```

Comment:
> please keep in mind we need to support non-POSIX systems where no poll() exists, like Windows.

There is a poll on windows. and it uses the same constants: https://msdn.microsoft.com/en-us/library/windows/desktop/ms740094(v=vs.85).aspx